From 5d53f80191f61f713311772afcc2ce08879942af Mon Sep 17 00:00:00 2001
From: Manikanta <mmaddireddy@nvidia.com>
Date: Tue, 27 Jun 2017 14:57:52 +0530
Subject: [PATCH 334/337] net: wireless: bcmdhd: fix warnings after applying
 rt-patches

Issue: Member type in struct completion is changed from
waitqueue_head_t to swait_queue_head in RT patches.

Fix: Use appropriate swait function call

bug 200320017

Signed-off-by: Manikanta <mmaddireddy@nvidia.com>
---
 drivers/net/wireless/bcmdhd/dhd_pno.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/dhd_pno.c b/drivers/net/wireless/bcmdhd/dhd_pno.c
index f85e873..be221c3 100644
--- a/drivers/net/wireless/bcmdhd/dhd_pno.c
+++ b/drivers/net/wireless/bcmdhd/dhd_pno.c
@@ -1550,7 +1550,7 @@ void * dhd_pno_get_gscan(dhd_pub_t *dhd, dhd_pno_gscan_cmd_cfg_t type,
 		_params->params_batch.get_batch.bytes_written = err;
 	}
 	mutex_unlock(&_pno_state->pno_mutex);
-	if (waitqueue_active(&_pno_state->get_batch_done.wait))
+	if (swait_active(&_pno_state->get_batch_done.wait))
 		complete(&_pno_state->get_batch_done);
 	return err;
 }
@@ -1996,7 +1996,7 @@ void * dhd_pno_get_gscan(dhd_pub_t *dhd, dhd_pno_gscan_cmd_cfg_t type,
 	{
 		struct dhd_pno_batch_params *params_batch;
 		params_batch = &_pno_state->pno_params_arr[INDEX_OF_BATCH_PARAMS].params_batch;
-		if (!waitqueue_active(&_pno_state->get_batch_done.wait)) {
+		if (!swait_active(&_pno_state->get_batch_done.wait)) {
 			DHD_PNO(("%s : WLC_E_PFN_BEST_BATCHING\n", __FUNCTION__));
 			params_batch->get_batch.buf = NULL;
 			params_batch->get_batch.bufsize = 0;
-- 
1.9.1

